Map display and simple movement logic

SiogunJH opened the pull request:
GridMap can now be displayed via GridManager.
GridEntities can now move through the GridMap

SiogunJH committed:
Grouped TileEntities

SiogunJH committed:
Tile Visualization

Placeholder 3D object representing a GridTile

SiogunJH committed:
Update SampleGridMap.asset

SiogunJH committed:
Transform Extensions

SiogunJH committed:
Update GridMap.cs

Added indexer and changed GridHeight/GridWidth to Height/Width

SiogunJH committed:
Grid Manager

SiogunJH committed:
Test Scene

Scene utilizaing GridManager

SiogunJH committed:
Merge branch 'dev' into feature/map-display

SiogunJH committed:
Enum separation

SiogunJH committed:
GridTile to GridTileData

SiogunJH committed:
GridTileController

MonoBehaviour with a reference to GridTileData

SiogunJH committed:
MonoBehaviourSingleton

SiogunJH committed:
Update GridMap.cs

Changed GridTile to GridTileData

SiogunJH committed:
OptimizeGrid function

Function that removes unnecessary GridTileData instances

SiogunJH committed:
Pathfinding

SiogunJH committed:
GridEntity as Serializable

SiogunJH committed:
GridManager as Singleton

SiogunJH committed:
Grid initialization precess changes

SiogunJH committed:
LogGridInfo

SiogunJH committed:
EntityID assignment logic

SiogunJH committed:
TileInitialization changes

SiogunJH committed:
Occupant initialization changes

SiogunJH committed:
UserFriendlyName derived from CharacterProfile

SiogunJH committed:
Location default value

SiogunJH committed:
Namespaces update

SiogunJH committed:
Map property

SiogunJH committed:
MonoBehaviour region

SiogunJH committed:
Move logic

SiogunJH committed:
MoveDebug logic

SiogunJH committed:
Update SampleGridMap.asset

SiogunJH committed:
GridEntity update

SiogunJH committed:
Update Tile Visualization (Placeholder).prefab

SiogunJH committed:
Update TestScene_Siogun.unity

AbexCorp commented on Assets/Scripts/Grid/GridTileData.cs:
```diff
@@ -0,0 +1,124 @@
+    public List<GridEntity> Occupants { get => _occupants == null ? new() : _occupants; }
+
+    //
+    public bool IsOccupied { get => _occupants != null && _occupants.Any(); } // this logic might need to be expanded in the future, as some GridEntities may be allowed to co-exist on the same tile (or not, idk)
```
There should be added a field to CharacterProfile or GridEntity to set if it occupies full tile, probably to GridEntity since its navigation related but the other is also fine

AbexCorp commented on Assets/Scripts/Grid/GridMap.cs:
```diff
@@ -12,39 +12,102 @@ public class GridMap : ScriptableObject
+
+    #region Pathfinding
+
+    public bool FindPathBetween(GridTileData start, GridTileData end, out IEnumerable<GridTileData> path)
```
We should decide if units from the same team can move through each other and make changes if needed

SiogunJH replied:
For now, they cannot - it might improve the strategic aspect of the game, it might not. Only testing will tell

AbexCorp commented on Assets/Scripts/Grid/GridEntity.cs:
```diff
@@ -1,15 +1,124 @@
+        }
+    }
+
+    private IEnumerator StepTo(GridTileData tile)
```
Entities that move on the grid change all needed references, but they remain as the children of the original tiles in hierarchy

AbexCorp commented on Assets/Scripts/Grid/GridEntity.cs:
```diff
@@ -1,15 +1,124 @@
+    #region DEBUG
+
+    [Button]
+    private void MoveAtRandom()
```
This is probably temporary but multiple calls cause teleportation

SiogunJH committed:
Tile occupation logic revisited

GridEntities can now be marked with "Occupies the Whole Tile" flag, and will pathfind accordingly

SiogunJH committed:
Update GridTileData.cs

Occupant-related methods now reparent occupants

SiogunJH committed:
Update GridEntity.cs

Restricted GridEntity to be able to handle only one move request at a time
